frontend/public/components/nav/section.tsx

```diff
@@ -104,30 +140,9 @@ export const NavSection = connect(navSectionStateToProps)(
       const { isOpen, activeChild } = this.state;
       const isActive = !!activeChild;
 
-      const mapChild = (c) => {
-        if (!c) {
-          return null;
-        }
-        const {name, required, disallowed} = c.props;
-        const requiredArray = required ? _.castArray(required) : [];
-        const requirementMissing = _.some(requiredArray, flag => (
-          flag && (flagPending(flags.get(flag)) || !flags.get(flag))
-        ));
-        if (requirementMissing) {
-          return null;
-        }
-        if (disallowed && (flagPending(flags.get(disallowed)) || flags.get(disallowed))) {
-          return null;
-        }
-        return React.cloneElement(c, {
-          key: name,
-          isActive: name === this.state.activeChild,
-          activeNamespace,
-          flags,
-        });
-      };
+      const getMapChild = c => mapChild(c, this.state.activeChild, flags, activeNamespace);
 
-      const Children = React.Children.map(children, mapChild);
+      const Children = React.Children.map(children, getMapChild);
```

Comment:
Note that [`React.Children.map`](https://reactjs.org/docs/react-api.html#reactchildrenmap) may return `null` or `undefined` if there are no children.

Consequently, `mergeElements` function should be prepared to handle this situation.

Reply:
done